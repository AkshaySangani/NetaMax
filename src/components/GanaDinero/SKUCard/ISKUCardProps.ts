export interface ISKUCardProps {
  /**
   * The card's title
   * @type {string}
   */
  title: string;

  /**
   * Product required number to complete the challenge
   * @type {number}
   */
  requiredItems: number;

  /**
   * Coins obtained when completing the challenge
   * @type {string | number}
   */
  awardCoins: string | number;

  /**
   * Boolean prop that changes the card to in progress state
   * @type boolean
   */
  accepted?: boolean;

  /**
   * Boolean prop that changes the card to completed state
   * @type boolean
   */
  completed?: boolean;

  /**
   * End date for the challenge
   * @type {Date}
   */
  endDate?: Date;

  /**
   * The progress bar value based on basket total
   * @type {number}
   */
  progressBarValue: number;

  /**
   * Button action to start the challenge
   * @type void function;
   */
  buttonAction: () => void;

  /**
   * Boolean that enables a different set of colors
   * @type boolean;
   */
  secondaryColor?: boolean;

  /**
   * Image to display
   * @type number
   */
  image?: string;
}
