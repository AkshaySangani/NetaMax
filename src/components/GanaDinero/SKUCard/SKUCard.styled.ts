import styled from "styled-components";

import { Progress } from "@chakra-ui/react";

export const CustomProgressBar = styled(Progress)`
  border-radius: 10px;
  background: linear-gradient(0deg, #412d73 0%, #25094f 100%);
  border: 2px solid #a80099;
  box-shadow: 0px 1px 1px rgba(93, 0, 108, 1);
  div:first-child {
    background: linear-gradient(0deg, #959800 0%, #ffe600 100%);
    height: 65%;
    border-radius: 10px;
    margin: 1.3px 0 0;
  }
`;
