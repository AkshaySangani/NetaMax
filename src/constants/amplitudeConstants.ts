export const HOME_VIEWED = "Home_Viewed";
export const PRODUCT_ADDED = "Product_Added";
export const CATEGORY_CLICKED = "Category_Clicked";
export const BANNER_CLICKED = "Banner_Clicked";
export const PRODUCT_DETAIL_VIEWED = "Product_Detail_Viewed";
export const CART_PAGE_VIEWED = "Cart_Page_Viewed";
export const NAME_ADDED = "Name_Added";
export const PHONE_ADDED = "Phone_Added";
export const ZIPCODE_PAGE_VIEWED = "Zipcode_Page_Viewed";
export const ZIPCODE_VERIFIED = "Zipcode_Verified";
export const LOGIN_COMPLETED = "Login_Completed";
export const ORDER_COMPLETED = "Order_Completed";
export const CHECKOUT_PAGE_VIEWED = "Checkout_Page_Viewed";
export const REFERRAL_CODE_CLICKED = "Referral_Code_Clicked";
export const CATEGORY_VER_TODOS_CLICKED = "Category_Ver_Todos_Clicked";
export const SEARCH_VIEW_MORE = "Search_View_More";
export const SEARCHBAR_CLICKED = "Searchbar_Clicked";
export const PROMO_SCROLLED = "Promo_Scrolled";
export const CATEGORY_SCROLLED = "Category_Scrolled";
export const BANNER_SCROLLED = "Banner_Scrolled";
export const CATEGORY_PRODUCT_SCROLLED = "Category_Product_Scrolled";
export const SUPPORT_CHAT_CLICKED = "Support_Chat_Clicked";
export const OTP_REQUESTED = "OTP_Requested";
export const OTP_PAGE_VIEWED = "OTP_Page_Viewed";
export const OTP_ERROR_TOASTED = "OTP_Error_Toasted";
export const OTP_RESEND_CODE = "OTP_Resend_Code";
export const STORE_SELECTED_CLICKED = "Store_Selected_Clicked";
export const SEARCH_ZIPCODE = "Search_ZipCode";
export const HOME_IS_YOUNGER_USER = "Home_Is_Younger_User";
export const NODE_PRODUCT_SCROLLED = "Node_Product_Scrolled";
export const NODE_VER_TODOS_CLICKED = "Node_Ver_Todos_Clicked";
export const HOME_ARROW_CLICKED = "Home_Arrow_Clicked";

// Admin shops event constants
export const SA_LOGIN_PAGE_VIEWED = "SA_Login_Page_Viewed";
export const SA_OTP_REQUESTED = "SA_otp_requested";
export const SA_LOGIN_COMPLETED = "SA_login_completed";
export const SA_OTP_TOAST_ERROR = "SA_otp_toast_error";
export const SA_LOGIN_FAILED = "SA_login_failed";
export const SA_INVOICES_PAGE_VIEWED = "SA_invoices_page_viewed";
export const SA_INVOICES_PAGE_SCROLLED = "SA_invoices_page_scrolled";
export const SA_INFO_PAGE_VIEWED = "SA_Info_Page_Viewed";
export const SA_SHARE_STORELINK_BANNER_CLICKED = "SA_Share_StoreLink_Banner_Clicked";
export const SA_CLIENTS_PAGE_VIEWED = "SA_Clients_Page_Viewed";
export const SA_PHONE_ADDED = "SA_Phone_Added";
export const SA_OTP_RESEND_CODE = "SA_OTP_Resend_Code";
export const SA_INVOICE_CLICKED = "SA_Invoice_Clicked";
export const SA_TRACK_LAST_ORDER_CLICKED = "SA_Track_Last_Order_Clicked";
//TODO: Check with business the grammar of 'Especific'
export const SA_TRACK_SPECIFIC_ORDER_CLICKED = "SA_Track_Especific_Order_Clicked";
export const SA_VIEW_ORDERS_CLICKED = "SA_View_Orders_Clicked";
export const SA_ORDERS_PAGE_VIEWED = "SA_Orders_Page_Viewed";
export const SA_VIEW_ORDER_DETAIL_CLICKED = "SA_View_Order_Detail_Clicked";
export const SA_ORDER_DETAIL_VIEWED = "SA_Order_Detail_Viewed";
export const SA_LOGOUT_CLICKED = "SA_Logout_Clicked";
export const SA_TOKEN_PAGE = "SA_Token_Page";
export const SA_TOKEN_REDEEMED = "SA_Token_Redeemed";
export const SA_ORDER_DELIVERED_PAGE = "SA_Order_Delivered_Page";
export const SA_TOKEN_ERROR_FOUND = "SA_Token_Error_Found";
export const SA_TOKEN_ALTERNATIVE_VALIDATION = "SA_Token_Alternative_Validation";
export const SA_FORCE_TOKEN_VALIDATION = "SA_Force_Token_Validation";
export const SA_DATE_SELECTED = "SA_Date_Selected";
export const SA_ACTUALIZAR_CLICKED = "SA_Actualizar_Clicked";
export const SA_COPIARLIGA_CLICKED = "SA_CopiarLiga_Clicked";
export const SA_HOME_PAGE_VIEWED = "SA_Home_Page_Viewed";
export const SA_VENTAS_PAGE_VIEWED = "SA_Ventas_Page_Viewed";
export const SA_ENDCLIENT_WA_CLICKED = "SA_EndClient_WA_Clicked";
export const SA_COMPARTIR_LIGA_CLICKED = "SA_Compartir_Liga_Clicked";
export const SA_TRACK_ORDER_CLICKED = "SA_Track_Order_Clicked";
export const SA_AYUDA_CLICKED = "SA_Ayuda_Clicked";
export const SA_VOLVER_BUTTON_CLICKED = "SA_Volver_Button_Clicked";
export const SA_DESCARGAR_BUTTON_CLICKED = "SA_Descargar_Button_Clicked";

// GanaDinero event constants
export const MONETAS_GMV_BUTTON_CLICKED = "Monetas_GMV_Quiero_comprar_Button_Clicked";
export const TAB_GANADINERO_BUTTON_CLICKED = "Tab_GanaDinero_Button_Clicked";
export const HOME_NO_MONETAS_BUTTON_CLICKED = "Home_No_Monetas_Button_Clicked";
export const HOME_MONETAS_NOT_LOGGED_BUTTON_CLICKED = "Home_Monetas_Not_Logged_Button_Clicked";
export const DAILY_REWARD_BUTTON_CLICKED = "Daily_Reward_Button_Clicked";
export const DAILY_REWARD_INGRESAR_DATOS_BUTTON_CLICKED =
  "Daily_Reward_Ingresar_Datos_Button_Clicked";
export const MONETAS_SKU_QUIERO_COMPRAR_BUTTON_CLICKED =
  "Monetas_SKU_Quiero_comprar_Button_Clicked";
export const MONETAS_GMV_PROGRESS_BAR_COMPLETED = "Monetas_GMV_Progress_Bar_Completed";
export const MONETAS_GMV_SUCCESS = "Monetas_GMV_Success";
export const MONETAS_SKU_BUTTON_CLICKED = "Monetas_Sku_Activated";
export const MONETAS_SKU_SUCCESS = "Monetas_SKU_Success";
export const MONETAS_SKU_PROGRESS_BAR_COMPLETED = "Monetas_SKU_Progress_Bar_Completed";
export const MONETAS_IS_SKU_CHALLENGE = "Is_Monetas_SKU_Challenge";
export const MONETAS_CATEGORY_BUTTON_CLICKED = "Monetas_Category_Activated";
export const MONETAS_CATEGORY_PROGRESS_BAR_COMPLETED = "Monetas_Category_Progress_Bar_Completed";
export const MONETAS_CATEGORY_SUCCESS = "Monetas_Category_Success";
export const MONETAS_IS_CATEGORY_CHALLENGE = "Is_Monetas_Category_Challenge";

// MisCompras event constants
export const TAB_MISCOMPRAS_BUTTON_CLICKED = "Tab_MisCompras_Button_Clicked";
export const TAB_MISCOMPRAS_VIDEO_PlAYED = "Tab_MisCompras_Video_Played";
export const TAB_MISCOMPRAS_IR_A_COMPRAR = "Tab_MisCompras_Ir_A_Comprar";
export const TAB_MISCOMPRAS_VER_DETALLE_PEDIDO = "Tab_MisCompras_Ver_Detalle_Pedido";
export const TAB_MISCOMPRAS_VER_DETALLE_PEDIDO_WHATSAPP_TIEND =
  "Tab_MisCompras_Ver_Detalle_Pedido_Whatsapp_Tiend";
export const TAB_MISCOMPRAS_VER_DETALLE_PEDIDO_VER_MAS =
  "Tab_MisCompras_Ver_Detalle_Pedido_Ver_Mas";

//MiCuenta event constants
export const MI_CUENTA_PAGE_VIEWED = "Mi_Cuenta_Page_Viewed";
export const MI_CUENTA_INGRESAR_DATOS_CLICKED = "Mi_Cuenta_Ingresar_Datos_Clicked";
export const MI_CUENTA_ERES_NUEVO_CLICKED = "Mi_Cuenta_Eres_Nuevo_Clicked";
export const MI_CUENTA_MIS_COMPRAS_PAGE_VIEWED = "Mi_Cuenta_Mis_Compras_Page_Viewed";
export const MIS_COMPRAS_NO_PURCHASES_PAGE_VIEWED = "Mis_Compras_No_Purchases_Page_Viewed";
export const MIS_COMPRAS_NO_PURCHASES_BUTTON_CLICKED = "Mis_Compras_No_Purchases_Button_Clicked";

// Checkout event constants
export const CHO_ENTER_ADDRESS_PAGE = "Enter_Address_Page";
export const CHO_STORES_RESULTS_LISTING_PAGE = "Stores_Results_Listing_Page";
export const CHO_ORDER_SUMMARY_PAGE = "Order_Summary_Page";

export enum TrackerApp {
  Amplitude = "amplitude",
  Segment = "segment",
}
