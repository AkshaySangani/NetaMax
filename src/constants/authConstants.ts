export const LOGIN_URL = "login/";
export const REFRESH_TOKEN_URL = "refresh/";
export const OTP_CODE_URL_SEC = "security/generateOtp";
export const GET_CUSTOMER_INFO_URL = "customer/me";
export const ZIP_CODE_URL = "zipcode_customer/";
export const OTP_VALIDATE_CODE_URL = "security/validateOtp";
export const INVALID_CODE = "Código inválido";
export const PLATFORM_OTP_LOGIN = "codigo_verificacion_usuario";
export enum OtpChannel {
  "WHATSAPP" = "whatsapp",
}
