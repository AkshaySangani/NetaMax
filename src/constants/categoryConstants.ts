export const GET_ALL_CATEGORIES_API_URL = "category/";
export const CATEGORY_TITLE = "Secciones";
export const DISCOUNT_TITLE = "Destacados";
export const MAX_CATEGORIES_PER_SLIDE_MOBILE = 2.3;
export const MAX_CATEGORIES_PER_SLIDE_TABLET = 4.3;
export const MAX_CATEGORIES_PER_SLIDE_DESKTOP = 6.3;
export const CATEGORY_SLIDE_SPACE_BETWEEN_ITEMS_MOBILE = 0;
export const CATEGORY_SLIDE_SPACE_BETWEEN_ITEMS_TABLET = 0;
export const CATEGORY_SLIDE_SPACE_BETWEEN_ITEMS_DESKTOP = 0;

export const categoryColors: Record<string, { color: string; name: string }> = {
  "70": { color: "#FFFDF5", name: "Lácteos y Refrigerados" },
  "73": { color: "#FFF9F9", name: "Canasta Básica" },
  "7": { color: "#FFFAEC", name: "Frutas y Verduras" },
  "3": { color: "#F3FFFC", name: "Despensa" },
  "68": { color: "#FFFDE9", name: "Licores y Cigarros" },
  "15": { color: "#F4F2FF", name: "Bebidas" },
  "69": { color: "#FAFFF2", name: "Dulces y Botanas" },
  "2": { color: "#FFFEF1", name: "Cuidado Personal" },
  "14": { color: "#FFEDED", name: "Mascotas" },
  "74": { color: "#FDFFF7", name: "Bebes y Niños" },
  "78": { color: "#F6FFFD", name: "Belleza" },
};

export const bannerNames = [
  "300.png",
  "301.png",
  "302.png",
  "305.png",
  "325.png",
  "327.png",
  "328.png",
];
