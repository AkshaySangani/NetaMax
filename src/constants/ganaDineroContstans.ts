export const GET_COINS_BY_CUSTOMER_ID = "customer-neta-wallet";

export const GMV_ID = 1;
export const EVENTS_ID = 2;
export const DISCOVERY_ID = 3;
export const REFERALS_ID = 4;
export const SURVEY_ID = 5;

// TODO: this might change when having real challenges code names or ids
export const SKU_NAME = "SKU";
export const CATEGORY_NAME = "Category";
