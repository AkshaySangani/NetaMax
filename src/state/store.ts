/* eslint-disable @typescript-eslint/no-explicit-any */
import {
  persistReducer,
  persistStore,
  FLUSH,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
  REHYDRATE,
} from "redux-persist";
import { createFilter } from "redux-persist-transform-filter";
import storage from "redux-persist/lib/storage";

import { authReducer } from "dataflows/Auth/AuthSlice";
import { basketReducer } from "dataflows/Basket/BasketSlice";
import { categoryReducer } from "dataflows/Category/CategorySlice";
import { checkoutReducer } from "dataflows/Checkout/CheckoutSlice";
import { eventTrackingReducer } from "dataflows/EventTracking/EventTrackingSlice";
import { ganaDineroReducer } from "dataflows/GanaDinero/GanaDineroSlice";
import { legalAgeReducer } from "dataflows/LegalAge/LegalAgeSlice";
import { misComprasReducer } from "dataflows/MisCompras/MisComprasSlice";
import { historialMonetasReducer } from "dataflows/Monetas/HistorialMonetasSlice";
import { nodeReducer } from "dataflows/Node/NodeSlice";
import { nopAuthReducer } from "dataflows/NopAuth/NopAuthSlice";
import { nopOrdersReducer } from "dataflows/OrderDetails/OrderSlice";
import { popupReducer } from "dataflows/Popup/PopupSlice";
import { categoryProductsReducer } from "dataflows/Product/CategoryProducts/CategoryProductsSlice";
import { megaPromosSectionReducer } from "dataflows/Product/MegaPromosSection/MegaPromosSectionSlice";
import { nodeProductsReducer } from "dataflows/Product/NodeProducts/NodeProductsSlice";
import { promotionProductsReducer } from "dataflows/Product/PromotionProducts/PromotionProductsSlice";
import { recommendedProductsReducer } from "dataflows/Product/RecommendedProducts/RecommendedProductsSlice";
import { searchReducer } from "dataflows/Product/Search/SearchSlice";
import { productReducer } from "dataflows/Product/SelectedProduct/SelectedProductSlice";
import { promotionReducer } from "dataflows/Promotion/PromotionSlice";
import { clientsReducer } from "dataflows/Shops/Clients/ClientsSlice";
import { deliverOrderReducer } from "dataflows/Shops/DeliverOrder/DeliverOrderSlice";
import { invoicesReducer } from "dataflows/Shops/Invoices/InvoicesSlice";
import { loginReducer } from "dataflows/Shops/Login/LoginSlice";
import { modalInvoiceReducer } from "dataflows/Shops/ModalInvoice/ModalInvoiceSlice";
import { ordersReducer } from "dataflows/Shops/Orders/OrdersSlice";
import { salesReducer } from "dataflows/Shops/Sales/SalesSlice";
import { shopInfoReducer } from "dataflows/Shops/ShopInfo/ShopsInfoSlice";
import { trackingReducer } from "dataflows/Shops/Tracking/TrackingSlice";
import { storeReducer } from "dataflows/Stores/IStoreSlice";

import { combineReducers, configureStore } from "@reduxjs/toolkit";

const saveSubsetAuthFilter = createFilter("auth", [
  "accessToken",
  "customer",
  "refreshToken",
  "deviceId",
]) as any;
const saveSubsetPopupFilter = createFilter("popup", ["popupShown"]) as any;
const saveSubsetGanaDineroFilter = createFilter("ganaDinero", [
  "coinsBeforeLogin",
  "isGMVNotified",
]) as any;

const rootPersistConfig = {
  key: "root",
  version: 2.1,
  storage,
  whitelist: ["basket", "auth", "popup", "ganaDinero", "legalAge", "store"],
  transforms: [saveSubsetAuthFilter, saveSubsetPopupFilter, saveSubsetGanaDineroFilter],
};

const shopsPersistConfig = {
  key: "login",
  version: 2.1,
  storage,
  whitelist: ["isAuth", "token", "shopUser"],
  blacklist: ["otpCodeError", "countDown", "currentStep"],
};

const rootReducer = combineReducers({
  auth: authReducer,
  basket: basketReducer,
  category: categoryReducer,
  products: combineReducers({
    categoryProducts: categoryProductsReducer,
    promotionProducts: promotionProductsReducer,
    megaPromosSection: megaPromosSectionReducer,
    nodeProducts: nodeProductsReducer,
    selectedProduct: productReducer,
    search: searchReducer,
    recommendedProducts: recommendedProductsReducer,
  }),
  monetas: historialMonetasReducer,
  checkout: checkoutReducer,
  store: storeReducer,
  promotion: promotionReducer,
  node: nodeReducer,
  popup: popupReducer,
  nopAuth: nopAuthReducer,
  nopOrders: nopOrdersReducer,
  tracking: trackingReducer,
  ganaDinero: ganaDineroReducer,
  misCompras: misComprasReducer,
  shops: combineReducers({
    shopInfo: shopInfoReducer,
    login: persistReducer(shopsPersistConfig, loginReducer),
    clients: clientsReducer,
    modalInvoice: modalInvoiceReducer,
    orders: ordersReducer,
    invoices: invoicesReducer,
    sales: salesReducer,
    deliverOrder: deliverOrderReducer,
  }),
  eventTracking: eventTrackingReducer,
  legalAge: legalAgeReducer,
});
const persistedReducer = persistReducer(rootPersistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export type NetaStore = typeof store;
