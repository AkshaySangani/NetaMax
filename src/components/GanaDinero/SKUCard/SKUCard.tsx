import React, { ReactElement } from "react";

import { Box, Button, Flex, Text, useMediaQuery, Image, Img, Spacer } from "@chakra-ui/react";

import { CardLayout } from "../CardLayout/CardLayout";
import { screenSizes } from "styled/screen";
import Check from "styled/icons/Check";
import { ISKUCardProps } from "./ISKUCardProps";
import { CustomProgressBar } from "./SKUCard.styled";
import { LANG_CODE } from "constants/localeFormatConstants";
import { BannerText, BannerWrapper, CardWrapper } from "../CardLayout/CardLayout.styled";

/**
 * SKUCard
 * @param {ISKUCardProps} props component props
 * @returns {ReactElement} the react element to render.
 */
export const SKUCard = (props: ISKUCardProps): ReactElement => {
  const {
    title,
    buttonAction,
    endDate,
    progressBarValue,
    requiredItems,
    secondaryColor,
    awardCoins,
    accepted,
    completed,
    image,
  } = props;
  const [isDesktop] = useMediaQuery(`(min-width: ${screenSizes.mdMin}px)`);

  if (completed) {
    return (
      <CardWrapper>
        <BannerWrapper>
          <BannerText data-testid="card-layout-banner">Te regalamos ${awardCoins}</BannerText>
        </BannerWrapper>
        <Box paddingTop="20px">
          <Box
            background="#E8E8E8"
            h="220px"
            boxShadow="0px 4px 4px rgba(0, 0, 0, 0.25)"
            borderRadius="12px"
            alignItems="center"
            display="flex"
            justifyContent="center"
            flexDirection="column"
            paddingTop="20px"
          >
            <Image src="/assets/images/icons/clock.png" alt="clock" />
            <Text margin="10px 20px" fontWeight="bold" fontSize="22px">
              ¡RETO COMPLETADO!
            </Text>
            <Text margin="0px 40px" textAlign="center" fontSize="18px">
              Vuelve mañana para seguir ganando monetas
            </Text>
          </Box>
        </Box>
      </CardWrapper>
    );
  }
  return (
    <CardLayout
      bannerText={`Te regalamos $${awardCoins}`}
      gradientColors={secondaryColor ? ["#8E35FF", "#7000FE"] : ["#B0009F", "#5B0077"]}
    >
      <Flex
        flexDirection="column"
        alignItems="center"
        justifyContent="center"
        pb="24px"
        h={isDesktop ? "335px" : "unset"}
      >
        <Flex
          alignItems="center"
          justifyContent="center"
          p="0 35px 0 25px"
          position="relative"
          flexDir={isDesktop ? "column" : "row"}
        >
          <Box mr={isDesktop ? 0 : "10px"} position="relative">
            <Flex
              backgroundColor="#FFF"
              borderWidth="3px"
              borderColor={secondaryColor ? "#7E19FE" : "#85008A"}
              w={isDesktop ? "35px" : "30px"}
              h={isDesktop ? "35px" : "30px"}
              borderRadius="99px"
              zIndex={2}
              position="absolute"
              alignItems="center"
              justifyContent="center"
              bottom="-18px"
              left={isDesktop ? "31%" : "25%"}
            >
              <Text
                color={secondaryColor ? "#7000CA" : "#A70098"}
                fontSize={isDesktop ? "12px" : "9px"}
                fontWeight="900"
                data-testid="sku-card-items"
              >
                {progressBarValue >= 100 ? (
                  <Check width={isDesktop ? "14px" : "12px"} height={isDesktop ? "14px" : "12px"} />
                ) : (
                  requiredItems
                )}
              </Text>
            </Flex>
            <Flex
              w={isDesktop ? "90px" : "60px"}
              h={isDesktop ? "90px" : "60px"}
              borderRadius="99px"
              overflow="hidden"
              position="relative"
              alignItems="center"
              justifyContent="center"
            >
              <Image src={image} zIndex={1} width={isDesktop ? "60px" : "50px"} mt="70px" />
              <Box
                backgroundColor="#C4C4C4"
                h={isDesktop ? "90px" : "60px"}
                w={isDesktop ? "90px" : "60px"}
                borderRadius={99}
                zIndex={0}
                mixBlendMode="hard-light"
                position="absolute"
              />
            </Flex>
          </Box>
          <Box>
            <Text
              fontSize={isDesktop ? "22px" : "20px"}
              fontWeight="bold"
              color="#FFF"
              mt={["0px", "20px"]}
              textAlign={isDesktop ? "center" : "left"}
              data-testid="sku-card-title"
            >
              {progressBarValue >= 100 ? "¡Llegaste a la meta!" : title}
            </Text>
            <Text
              fontSize="14px"
              color="#FFF"
              fontWeight="500"
              textAlign={isDesktop ? "center" : "left"}
              data-testid="sku-card-description"
            >
              {progressBarValue >= 100
                ? `Finaliza tu pedido actual y gana tus ${awardCoins} monetas.`
                : `Ganas ${awardCoins} monetas para tu próxima compra.`}
            </Text>
          </Box>
        </Flex>
        {accepted ? (
          <Box p="0 35px 0 25px" mt="20px" w="100%">
            <Box className="progress-bar" pos="relative" w={["98%", "100%"]}>
              <CustomProgressBar min={0} max={requiredItems} value={progressBarValue} />
              <Img
                src={`/assets/images/icono_gana_${secondaryColor ? "2" : "1"}.png`}
                height="31px"
                pos="absolute"
                right="-15px"
                top="-10px"
              />
            </Box>
            {progressBarValue < 100 && (
              <Flex mt="12px">
                <Box>
                  <Text
                    fontSize={isDesktop ? "14px" : "10px"}
                    fontWeight="400"
                    color="rgba(255, 255, 255, .5)"
                  >
                    Lejos de ganar
                  </Text>
                </Box>
                <Spacer />
                <Box>
                  <Text
                    fontSize={isDesktop ? "14px" : "10px"}
                    fontWeight="700"
                    color="rgba(255, 255, 255, .5)"
                  >
                    GANASTE
                  </Text>
                </Box>
              </Flex>
            )}
            <Text
              fontSize="14px"
              color="#FFF"
              fontWeight="500"
              data-testid="sku-card-date"
              mt="15px"
            >
              *Válido hasta el{" "}
              {endDate?.toLocaleDateString(LANG_CODE, {
                month: "long",
                day: "numeric",
              })}
            </Text>
          </Box>
        ) : (
          <Flex justifyContent="center" alignItems="center" mt="28px">
            <Text fontSize="14px" color="#FFF" fontWeight="500" w="30%" data-testid="sku-card-date">
              *Válido hasta el{" "}
              {endDate?.toLocaleDateString(LANG_CODE, {
                month: "long",
                day: "numeric",
              })}
            </Text>
            <Button
              onClick={buttonAction}
              borderRadius="16px"
              h="56px"
              w="154px"
              backgroundColor="#FFF"
              color="#3870FF"
              _hover={{ bgColor: "none" }}
              data-testid="SKU-card-button"
              ml="10px"
            >
              Empezar reto
            </Button>
          </Flex>
        )}
      </Flex>
    </CardLayout>
  );
};
