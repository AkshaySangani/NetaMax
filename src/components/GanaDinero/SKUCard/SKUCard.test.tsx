import { render, cleanup } from "@testing-library/react";

import { SKUCard } from "./SKUCard";
import { ISKUCardProps } from "./ISKUCardProps";
import { SKU_NAME } from "constants/ganaDineroContstans";
import { Provider } from "react-redux";
import { store } from "state/store";

describe("Render SKUCard component", () => {
  const commonProps: ISKUCardProps = {
    title: "Test product title",
    requiredItems: 5,
    endDate: new Date(2022, 4, 28),
    progressBarValue: 0,
    buttonAction: jest.fn(),
    awardCoins: 10,
    type: SKU_NAME,
  };

  // TODO: ADD ADDITIONAL TEST CASES WHEN THE COMPONENT RECEIVE DATA FROM TEH ENDPOINT AND HAVE ADDITIONAL STATES
  test("SKUCard should render properly when having coins to claim", () => {
    const { container } = render(
      <Provider store={store}>
        <SKUCard {...commonProps} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    cleanup();
  });
});
