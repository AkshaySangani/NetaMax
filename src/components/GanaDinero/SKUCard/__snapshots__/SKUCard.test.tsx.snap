// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SKUCard component SKUCard should render properly when having coins to claim 1`] = `
<div>
  <div
    class="sc-bdvvtL bvVYQj css-k008qs"
  >
    <div
      class="sc-gsDKAQ kWXsUD css-0"
    >
      <div
        class="sc-dkPtRN egXpyz css-0"
        data-testid="card-layout-banner"
      >
        Te regalamos $10
      </div>
    </div>
    <div
      class="css-7w6khc"
    >
      <div
        class="css-10klw3m"
      >
        <div
          class="sc-hKwDye dlqxDY css-0"
        >
          <div
            class="sc-eCImPb ftCsbL css-101gstn"
            data-testid="card-layout-children"
          >
            <div
              class="css-10vksws"
            >
              <div
                class="css-zyhhhw"
              >
                <div
                  class="css-15foczu"
                >
                  <div
                    class="css-2nkt43"
                  >
                    <p
                      class="chakra-text css-1mnrwgd"
                      data-testid="sku-card-items"
                    >
                      5
                    </p>
                  </div>
                  <div
                    class="css-50t65g"
                  >
                    <img
                      class="chakra-image css-1hihhg0"
                    />
                    <div
                      class="css-kkn1gn"
                    />
                  </div>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-t7zs25"
                    data-testid="sku-card-title"
                  >
                    Test product title
                  </p>
                  <p
                    class="chakra-text css-40gfk0"
                    data-testid="sku-card-description"
                  >
                    Ganas 10 monetas para tu próxima compra.
                  </p>
                </div>
              </div>
              <div
                class="css-1aljbdn"
              >
                <p
                  class="chakra-text css-1v8sl5v"
                  data-testid="sku-card-date"
                >
                  *Válido hasta el
                   
                  28 de mayo
                </p>
                <button
                  class="chakra-button css-t905fy"
                  data-testid="SKU-card-button"
                  type="button"
                >
                  Empezar reto
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
